src/api/zoops.ts

```diff
@@ -55,17 +55,30 @@ zoopsRouter.post("/", requireUser, async (req, res, next): Promise<void> => {
 
 // PUT /api/zoops/:id
 zoopsRouter.put("/:id", requireUser, async (req, res, next)=> {
-    try {
-        const {id} = req.params;
-        const {content} = req.body;
-        const zoop = await prisma.zoop.update({
-            where: {id: Number(id)},
-            data: {content}
-        })
-        res.send({zoop});
-    } catch (e) {
-        next(e);
+    const userId = req.user?.id;
+    const {id} = req.params;
+
+    const zoop = await prisma.zoop.findUniqueOrThrow({
+        where: {id: Number(id)}
+    });
+
+    if (zoop.authorId === userId) {
```

Comment:
nit: You could also send 404 to front-end if there isn't a zoop with that ID, rather than throwing

```suggestion
    const zoop = await prisma.zoop.findUnique({
        where: {id: Number(id)}
    });
   if (!zoop) {
     res.status(404);
     next({name: "NotFound", message: "Zoop with ID not found"});
   } else if (zoop.authorId === userId) {
```